docs/submission/submitting-molecular-data.md

```diff
@@ -57,78 +57,161 @@ storage.url=https://score.qa.argo.cancercollaboratory.org
+
+If the information for a sample is different than what has been registered, validation will fail immediately upon submission.
+
+**Read Groups:** The read group section contains details about the reads that were generated from a single run of a sequencing instrument lane. The number of `read_group` objects in the payload must meet the number specified in `read_group_count`.
+
+- `**submitter_read_group_id`: The unique identifier of a read group.
+- `**platform_unit`:
+- `**is_paired_end`: `true` for paired end sequencing, otherwise `false`.
+- `**file_r1`: Name of the read group file to be submitted.
+- `file_r2`: Name of the read group file to be submitted. Required if and only if paired end sequencing was done.
+- `read_length_r1`: Average length of reads in `file_r1`.
+- `read_length_r2`: Average length of reads in `file_r2`.
+- `insert_size`:
+- `sample_barcode`:
+- `**library_name`:
+
+Read Group Data Validations:
+
+1. `submitter_read_group_id` must be unique in within each Song payload, and ideally unique across all read groups in an ARGO program.
+1. `submitter_read_group_id` must not contain any special characters, with the exception of `-` , `.` , and `\_` .
+1. All `read_groups` in the payload must belong to a single sample.
+1. `platform_units` must be unique in metadata with a one-to-one relationship with `submitter_read_group_id`.
+1. The total number of `read_group` objects must match the number specified in `read_group_count`.
+1. For paired end sequencing, both `file_r1` and `file_r2` are required, otherwise, only `file_r1` is required (`file_r2` must not be populated).
+1. For FASTQ submission, no file can appear more than once in `file_r1` or `file_2`.
+
+**Files:** The files section contains metadata about the molecular files to be submitted. If multiple read groups were sequenced, then multiple files should be listed as objects in the payload.
+
+ARGO accepts sequencing data submission for both `BAM` and `FASTQ` files. There is no special requirement for FASTQ files except that paired end data should have the reads in two FASTQ files (one for each end). ARGO does not accept interleaved FASTQ files. We encourage compression on FASTQ files; both _gzip_ (suffix .fq.gz or .fastq.gz) or _bz2_ (suffix .fq.bz2 or .fastq.bz2) are supported.
+
+- `**fileName`: Name of the file, as defined by the data submitter.
+- `**fileSize`: Provided in bytes.
+- `**fileMd5sum`: Compute the md5sum of the file. This must match what is computed when the file is uploaded.
+- `**fileType`: Set to `BAM` or `FASTQ` based file type being submitted.
+- `**fileAccess`: Set to `controlled`.
+- `**dataType`: Set to `submitted_reads`.
+
+For both FASTQ and BAM submission, all files in `files` section must be unique.

```

Comment:
For both FASTQ and BAM `submissions`, all files in `the` `files` section must be unique.